File: convert.js
const fs = require('fs');

// 1. Đọc file SQL
const sqlContent = fs.readFileSync('data.sql', 'utf8');

// 2. Trích xuất các giá trị trong câu lệnh INSERT INTO
const regex = /\((\d+),\s*'([^']+)',\s*'([^']*)'\)/g;
const records = [];
let match;

while ((match = regex.exec(sqlContent)) !== null) {
  records.push({
    stt: parseInt(match[1], 10),
    page: match[2],
    keyword: match[3]
  });
}

// 3. Ghi ra file db.js
const fileContent = `const database = ${JSON.stringify(records, null, 2)};\n\nexport default database;\n`;
fs.writeFileSync('db.js', fileContent, 'utf8');

console.log(`Thành công! Đã chuyển đổi ${records.length} bản ghi sang db.js.`);

File: convert.py
import re
import json

# 1. Tên file SQL chính xác theo thư mục của bạn
sql_file_path = 'db_trichdan.sql'
js_file_path = 'db.js'

# 2. Đọc file SQL
with open(sql_file_path, 'r', encoding='utf-8') as f:
    sql_content = f.read()

# 3. Regex tìm cấu trúc: (stt, 'page', 'keyword')
pattern = r"\(\s*(\d+)\s*,\s*'([^']*)'\s*,\s*'([^']*)'\s*\)"
matches = re.findall(pattern, sql_content)

# 4. Trích xuất dữ liệu
data = []
for match in matches:
    data.append({
        "stt": int(match[0]),       # match[0] mới là số stt
        "page": match[1],      # match[1] là tên trang (tr0001.png)
        "keyword": match[2]   # match[2] là từ khóa
    })

# 5. Xuất ra file JS
js_content = f"const database = {json.dumps(data, ensure_ascii=False, indent=2)};\n\nexport default database;\n"

with open(js_file_path, 'w', encoding='utf-8') as f:
    f.write(js_content)

print(f"Thành công! Đã chuyển đổi {len(data)} bản ghi sang file {js_file_path}.")